jaworski_cellseg/configuration_widget.py

```diff
@@ -0,0 +1,151 @@
+from pathlib import Path
+
+def create_configuration_widget(
+    configurations: list[str], current_config: int, config_file_path: Path, jaworski_obj
+):
+    """
+    Widget to save and load values from the GUI
+    """
+
+    def generate_configuration_data(jaworski_obj):
+        try:
+            # get configuration values
+            data = {
+                "gausianFilter": jaworski_obj.pre_process_data_widget.gaussian_checkbox.value,
+                "gaussian_sigma": jaworski_obj.pre_process_data_widget.gaussian_factor.value,
+                "contrast_adjustment": jaworski_obj.pre_process_data_widget.contrast_adjustment.value,
+                "contrast_min": jaworski_obj.pre_process_data_widget.contrast_lower.value,
+                "contrast_max": jaworski_obj.pre_process_data_widget.contrast_upper.value,
+                "binary_map_threshold": jaworski_obj.pre_process_data_widget.binary_map_threshold.value,
+                "inference_use_window_choice": jaworski_obj.inferer_widget.use_window_choice.isChecked(),
+                "inference_window_size": int(
+                    jaworski_obj.inferer_widget.window_size_choice.currentText()
+                ),
+                "inference_input": jaworski_obj.inferer_widget.model_input_size.value(),
+                "inference_overlap": jaworski_obj.inferer_widget.window_overlap_slider.value(),
+                "inference_perform_threhold": jaworski_obj.inferer_widget.thresholding_checkbox.isChecked(),
+                "inference_perform_threhold_value": jaworski_obj.inferer_widget.thresholding_slider.value(),
+                "inference_instance_segmentation": jaworski_obj.inferer_widget.use_instance_choice.isChecked(),
+                "inference_instance_segmentation_option": "Voronoi-Otsu",
+            }
+
+            # Voronoi-Otsu items
+            voronoi_widget = jaworski_obj.inferer_widget.instance_widgets.methods[
+                "Voronoi-Otsu"
+            ]
+            data.update(
+                {
+                    "inference_instance_segmentation_spot_signma": voronoi_widget.counters[
+                        0
+                    ].value(),
+                    "inference_instance_segmentation_outline_signma": voronoi_widget.counters[
+                        1
+                    ].value(),
+                    "inference_instance_segmentation_small_object_removal": voronoi_widget.counters[
+                        2
+                    ].value(),
+                }
+            )
+
+            return data
+        except AttributeError as e:
+            print(f"Error accessing configuration values: {e}")
+            return {}
+
+    def set_widget_values(widget, config, mappings):
+        try:
+            for attr, key in mappings.items():
+                # Split the attribute path
+                parts = attr.split(".")
+                obj = widget
+                for part in parts[:-1]:
+                    obj = getattr(obj, part)
```

Comment:
I'm a bit confused by this. Can you say a word or two about what you're trying to do here? If you want the second-to-last member of `parts` you can select it with `parts[-2]`, no for loop needed.

Reply:
Done! I applied the `parts[-2]`